### PR TITLE
Fixes Cyborgs being labeled as Staff assistants by cryogenic sleep units.

### LINKED ISSUE #10883: Entering cryogenic storage as Cyborg instead reads as Staff Assistant when entering it
### Labels

[BUG][TRIVIAL]

### Description

as per title.

### Steps to reproduce

1. Enter cyro as a cyborg, for whatever reason
2. see "[Character name] the Staff Assistant has entered cryogenic storage."
3. get slightly confused

### Expected Behavior

for it to read "[Character name] the Cyborg has entered cryogenic storage.", when youre a borg of course.

### Additional Information & Screenshots

Reported by: Musrom
On server: Goonstation 3 Roleplay: Morty
Round log date: 2022-09-18-20-04
Reported on: 2022-09-18 21:06:20

### PR DESCRIPTION
**Pull request by Debi-sys**
## About the PR 
It fixes the trivial bug of cyborgs being labeled as staff assistants by cryo sleep units

## Why's this needed? 
Annoying, seems off and fixes #10883

### CODE REVIEW

**Comment by Debi-sys** [comment]
Excuse the other contributor, accidently merged with my alternative account.

File: code/obj/machinery/computer/announcement.dm [hunk 1]
```diff
@@ -225,9 +225,12 @@
 		var/job = person.mind.assigned_role
 		if(!job || job == "MODE")
 			job = "Staff Assistant"
+		if(ishivebot(person) || isrobot(person))
```
Comment (TobleroneSwordfish) [inline code/obj/machinery/computer/announcement.dm:228]: Could probably just be `issilicon`?

Comment (Debi-sys) [reply code/obj/machinery/computer/announcement.dm:228]: Testing it right now.

Comment (Debi-sys) [reply code/obj/machinery/computer/announcement.dm]: Yeah that worked, simplified it and synced changed.

**Comment by ZWRh3** [comment]
can AI enter cryo?

**Comment by Debi-sys** [comment]
> can AI enter cryo? if yes it will get mislabeled
I'm testing it right now.

**Comment by Debi-sys** [comment]
> can AI enter cryo? if yes it will get mislabeled

AI cannot enter  cryo, trying to do so results in "An incorporeal manifestation of an artificial intelligence's presence can't enter the industrial cryogenic sleep unit!"

File: code/obj/machinery/computer/announcement.dm [hunk 2]
```diff
@@ -225,9 +225,12 @@
 		var/job = person.mind.assigned_role
 		if(!job || job == "MODE")
 			job = "Staff Assistant"
+		if(issilicon(person))
+			job = "Cyborg"
```
Comment (ZeWaka) [inline code/obj/machinery/computer/announcement.dm:229]: Do cyborgs not have an assigned role?

Comment (Debi-sys) [reply code/obj/machinery/computer/announcement.dm:229]: Not sure to be fair, but this was my first try at repairing a bug and i mostly followed instructions of people from the discord server. In the end i was resulted with this, tested it and it worked. 😄

**Comment by frawhst** [comment]
I think what might have happened is that someone who was a staff assistant at the start of the round (thus, `assigned_role` of SA) was borged, and when entering cryo they were announced as an assistant. The same thing would probably happen for say, a security officer borged- they'd be called a security officer when cryoing, even though they're a robot.
I'd expect roundstart cyborgs to work fine, since they do have an assigned_role of cyborg.

tldr yeah robots should say 'cyborg' regardless of assigned_role so this is good imo

**Review by frawhst — APPROVED** [review]
